src/Writers/LaTeXWriter.jl

```diff
@@ -83,23 +84,64 @@ function render(doc::Documents.Document)
                 writefooter(context, doc)
             end
             cp(STYLE, "documenter.sty")
-            if hastex()
-                outdir = joinpath(doc.user.root, doc.user.build)
-                pdf = replace("$(doc.user.sitename).pdf", " " => "")
-                try
-                    run(`latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $file`)
-                catch err
-                    @warn "failed to compile. Check generated LaTeX file."
-                    cp(file, joinpath(outdir, file); force = true)
-                end
-                cp(pdf, joinpath(outdir, pdf); force = true)
-            else
-                @warn "`latexmk` and `lualatex` required for PDF generation."
-            end
+
+            # compile .tex and copy over the .pdf file if compile_tex return true
+            status = compile_tex(texfile)
+            status && cp(pdffile, joinpath(doc.user.root, doc.user.build, pdffile); force = true)
         end
     end
 end
 
+const DOCKER_IMAGE_TAG = "0.1"
+
+function compile_tex(texfile)
+    engine = get(ENV, "DOCUMENTER_LATEX_ENGINE", "latexmk") # TODO: make this configurable from makedocs
```

Comment:
Having a `Documenter.LaTeX` plugin for this would be good. For simplicity, we could also implement it like `Documenter.HTML` for now, and change to `format = Documenter.HTML(...)` or `writer = Documenter.HTML(...)` later.